internal/pkg/manifest/interpolate.go

```diff
@@ -0,0 +1,67 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package manifest
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+)
+
+const (
+	predefinedEnvVarKeyForAppName = "COPILOT_APPLICATION_NAME"
+	predefinedEnvVarKeyForEnvName = "COPILOT_ENVIRONMENT_NAME"
+)
+
+var (
+	// Modified from docker/compose
+	interpolatorRegexp = regexp.MustCompile(`\${([_a-zA-Z][_a-zA-Z0-9]*)}`)
+)
+
+type interpolator struct {
+	predefinedEnvVars map[string]string
+}
+
+type predefinedEnvVar struct {
+	appName string
+	envName string
+}
+
+func newInterpolator(envVars predefinedEnvVar) *interpolator {
+	return &interpolator{
+		predefinedEnvVars: map[string]string{
+			predefinedEnvVarKeyForAppName: envVars.appName,
+			predefinedEnvVarKeyForEnvName: envVars.envName,
+		},
+	}
+}
+
+func (i *interpolator) substitute(s string) (string, error) {
+	matches := interpolatorRegexp.FindAllStringSubmatch(s, -1)
+	if len(matches) == 0 {
+		return "", nil
+	}
+	replaced := s
+	for _, match := range matches {
+		// https://pkg.go.dev/regexp#Regexp.FindAllStringSubmatch
+		key := match[1]
+		subTextRegexp := regexp.MustCompile(fmt.Sprintf(`\${%s}`, key))
```

Comment:
What do you think of renaming it as:
```suggestion
		currSegmentRegex := regexp.MustCompile(fmt.Sprintf(`\${%s}`, key))
```

Reply:
Do we need this to be a regexp? 🥺 can't we do something along the lines:
```
strings.ReplaceAll(s, fmt.Sprintf("${%s}", key), os.Getenv(key)) 
```